python/dgl/nn/pytorch/sparse_emb.py

```diff
@@ -187,15 +169,13 @@ def __call__(self, node_ids, device=th.device("cpu")):
         device : th.device
             Target device to put the collected embeddings.
         """
-        if not self._comm or self._comm.size() == 1:
+        if not self._comm:
             emb = self._tensor[node_ids].to(device)
         else:
-            if self.world_size > 0:
-                emb = self._comm.sparse_all_to_all_pull(
-                    node_ids, self._tensor, self._partition
-                )
-            else:
-                emb = self._tensor[node_ids]
+            # will handle self._world_size = 0 or 1
```

Comment:
This comment is not very clear to me.
What will handle self._world_size = 0 or 1? What will happen if self._world_size not equals 0 or 1?

Can you elaborate more?

Reply:
The following `nccl.sparse_all_to_all_pull` will handle  self._world_size = 0 or 1, where we can just return `self._tensor[node_ids]`.